full-moon/src/ast/visitors.rs

```diff
@@ -370,8 +370,20 @@ impl Visit for LocalAssignment {
         visitor.visit_local_assignment(self);
         self.local_token.visit(visitor);
 
+        let mut attributes;
```

Comment:
You shouldn't need mut here (you don't need it on type_specifiers either).

Reply:
Its necessary because `attributes` and `type_specifiers` are iterators which we mutate by calling `.next()` on later
https://github.com/Kampfkarren/full-moon/blob/aae3a2339bdbe3d677cd3f60775ed88aadb49f25/full-moon/src/ast/visitors.rs#L398-L402